src/index.ts

```diff
@@ -68,13 +68,24 @@ function isMergeableObject <T> (val: T): boolean {
 export function merge <T extends object, U extends object> (obj: T, src: U): T & U {
   const res = obj as T & U
   for (const key in src) {
-    const clone = isMergeableObject(src[key]) ? cloneDeep(src[key]) : undefined
+    // `has` check needed to avoid using inherited properties. This is probably
+    // closer to the behaviour one expects and also avoids prototype pollution.
+    const clone = has(obj, key) && isMergeableObject(src[key]) ? cloneDeep(src[key]) : undefined
```

Comment:
So if `obj` doesn't have a key that `src` does have, it won't merge that value in? That seems wrong..

Reply:
I'll merge the object in, but you're right that this is a bug because a clone of src[key] should be used. The correct location to do this check is (was) below to avoid merging on a property that `obj` doesn't have (like `obj.__proto__` or `obj.constructor`)